File: py101-basics/exercises/easy_3/bannerizer.py
def one_line_fit(string):
    print(f"+-{''.join('-' for _ in string)}-+")
    print(f"| {''.join(' ' for _ in string)} |")
    print(f"| {string} |")
    print(f"| {''.join(' ' for _ in string)} |")
    print(f"+-{''.join('-' for _ in string)}-+")

def print_in_box(string, max_width=None):
    char_list = list(string)
    if max_width:
        if max_width < len(string):
            print(f"+-{''.join('-' * max_width)}-+")
            print(f"| {''.join(' ' * max_width)} |")
            print(f"| {''.join(char_list[:max_width])} |")
            print(f"| {''.join(' ' * max_width)} |")
            print(f"+-{''.join('-' * max_width)}-+")
        else:
            one_line_fit(string)
    else:
        one_line_fit(string)

print_in_box('To boldly go where no one has gone before.', 40)
print_in_box('To boldly go where no one has gone before.', 5)
print_in_box('To boldly go where no one has gone before.', 500)
print_in_box('To boldly go where no one has gone before.')

File: js210/exercises/easy_2/bannerizer.js
/*
- A
while count is less than 5
  if count is equal to 0 or 4
    log to console '+-' + '-' multiplied by length of string + '-+'
  if count is equal to 1 or 3
    log to console '| ' + ' ' multiplied by length of string + ' |'
  if count is equal to 2
    log to console '| ' + string + ' |'  

*/

function buildWidth(char, amount) {
  let result = '';
  for (let i = 0; i < amount; i++) {
    result = result + char;
  }
  return result;
}


function logInBox(string) {
  let length = string.length;
  let count = 0;
  while (count < 5) {
    if (count === 0 || count === 4) {
      console.log(`+-${buildWidth('-', length)}-+`);
    } else if (count === 1 || count === 3) {
      console.log(`| ${buildWidth(' ', length)} |`)
    } else {
      console.log(`| ${string} |`)
    }
    count++;
  }
}

logInBox('To boldly go where no one has gone before.');
console.log();
logInBox('');